File: 1768-merge-strings-alternately/1768-merge-strings-alternately.cpp
class Solution {
public:
    string mergeAlternately(string word1, string word2) {
        string res = "";
        int p1 = 0, p2 = 0;

        while(p1<word1.size() && p2<word2.size()){
            res = res + word1[p1] + word2[p2];
            p1++;
            p2++;
        }

        return p1 == word1.size()? res+word2.substr(p2, word2.size()-p2): res+word1.substr(p1, word1.size()-p1);
        
    }
};

File: 1768-merge-strings-alternately/1768-merge-strings-alternately.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        m = len(word1)
        n = len(word2)
        result = []
        i , j = 0, 0

        while i < m or j < n:
            if i < m:
                result += word1[i]
                i += 1

            if j < n:
                result += word2[j]
                j += 1

        return "".join(result)
        